cmd/evm/runner.go

```diff
@@ -274,24 +283,32 @@ func runCmd(ctx *cli.Context) error {
 	if ctx.Bool(CreateFlag.Name) {
 		input = append(code, input...)
 		execFunc = func() ([]byte, uint64, error) {
+			// don't mutate the state!
+			runtimeConfig.State = prestate.Copy()
 			output, _, gasLeft, err := runtime.Create(input, &runtimeConfig)
 			return output, gasLeft, err
 		}
 	} else {
 		if len(code) > 0 {
-			statedb.SetCode(receiver, code)
+			prestate.SetCode(receiver, code)
 		}
 		execFunc = func() ([]byte, uint64, error) {
+			// don't mutate the state!
+			runtimeConfig.State = prestate.Copy()
 			output, gasLeft, err := runtime.Call(receiver, input, &runtimeConfig)
 			return output, initialGas - gasLeft, err
 		}
 	}
 
 	bench := ctx.Bool(BenchFlag.Name)
-	output, stats, err := timedExec(bench, execFunc)
+	output, stats, execErr, benchErr := timedExec(bench, execFunc)
+	if benchErr != nil {
+		fmt.Printf("benchmarking execution failed: %v\n", benchErr)
+		return benchErr
+	}
```

Comment:
Why not something a bit more go-idiomatic?
```suggestion
	output, stats, execErr, err := timedExec(bench, execFunc)
	if err != nil {
		return fmt.Errorf("benchmarking failed: %v\n", err)
	}
```